services/kubernetesmisc/src/handlers/kubernetesBuildCancel.ts

```diff
@@ -0,0 +1,135 @@
+import * as R from 'ramda';
+import { getOpenShiftInfoForProject, updateTask, updateDeployment } from '@lagoon/commons/src/api';
+import { logger } from "@lagoon/commons/src/local-logging";
+import { sendToLagoonLogs } from '@lagoon/commons/src/logs';
+
+import Api, { ClientConfiguration } from 'kubernetes-client';
+const Client = Api.Client1_13;
+
+const ocsafety = string => string.toLocaleLowerCase().replace(/[^0-9a-z-]/g, '-');
+
+const generateSanitizedNames = (project, environment, projectInfo) => {
+  try {
+    const safeBranchName = ocsafety(environment.name);
+    const safeProjectName = ocsafety(project.name);
+    const namespace = projectInfo.openshiftProjectPattern
+      ? projectInfo.openshiftProjectPattern
+          .replace('${branch}', safeBranchName)
+          .replace('${project}', safeProjectName)
+      : `${safeProjectName}-${safeBranchName}`;
+    return { namespace, safeProjectName };
+  } catch (error) {
+    logger.error(`Error while loading information for project ${project.name}`);
+    logger.error(error);
+    throw error;
+  }
+};
+
+const getUrlTokenFromProjectInfo = (projectOpenShift, name) => {
+  try {
+    const url = projectOpenShift.openshift.consoleUrl.replace(/\/$/, '');
+    const token = projectOpenShift.openshift.token || '';
+    return { url, token };
+  } catch (error) {
+    logger.warn(
+      `Error while loading information for project ${name}: ${error}`
+    );
+    throw error;
+  }
+};
+
+const getConfig = (url, token) => ({
+  url,
+  insecureSkipTlsVerify: true,
+  auth: {
+    bearer: token
+  }
+});
+
+const getJobInfo = async (client: Api.ApiRoot, namespace: string, jobName: string) => {
+  let result;
+  try {
+    result = await client.apis.batch.v1.namespaces(namespace).jobs(jobName).get()
+  } catch (err) {
+    if (err.code == 404) {
+      return undefined;
+    } else {
+      logger.error(err);
+      throw new Error();
+    }
+  }
+  return result;
+}
+
+const updateLagoonTask = async (jobInfo, jobStatus, taskId, project, jobName) => {
```

Comment:
I thought I saw that you deleted this function in a commit? But either way, it's not being used at all, so it can go

Reply:
👍